src/library/scala/util/hashing/MurmurHash3.scala

```diff
@@ -159,18 +212,76 @@ private[hashing] class MurmurHash3 {
     finalizeHash(h, data.length)
   }
 
+  /** Compute the hash of an IndexedSeq. Potential range hashes are recognized to produce a
+    * hash that is compatible with rangeHash.
+    */
+  final def indexedSeqHash(a: scala.collection.IndexedSeq[Any], seed: Int): Int = {
+    var h = seed
+    val l = a.length
+    l match {
+      case 0 =>
+        finalizeHash(h, 0)
+      case 1 =>
+        finalizeHash(mix(h, a(0).##), 1)
+      case _ =>
+        val initial = a(0).##
+        h = mix(h, initial)
+        val h0 = h
+        var prev = a(1).##
+        val rangeDiff = prev - initial
+        var i = 2
+        while (i < l) {
```

Comment:
This might be faster if the loop condition included a test for `rangeDiff != hash - prev` and the still-have-`i < l` case is detected afterwards.  I'm not sure whether you tested it both ways, but usually small loops optimize better.

Reply:
`hash` is computed inside the loop. I didn't find a faster way of doing this.